pkgs/phps.nix

```diff
@@ -20,100 +13,94 @@ let
-            ];
-
-          buildInputs =
-            attrs.buildInputs
-            ++ prev.lib.optionals (prev.lib.versionOlder args.version "7.1") [
-              prev.libxcrypt
-            ];
-
-          preConfigure =
-            prev.lib.optionalString (prev.lib.versionOlder args.version "7.4") ''
-              # Workaround “configure: error: Your system does not support systemd.”
-              # caused by PHP build system expecting PKG_CONFIG variable to contain
-              # an absolute path on PHP ≤ 7.4.
-              # Also patches acinclude.m4, which ends up being used by extensions.
-              # https://github.com/NixOS/nixpkgs/pull/90249
-              for i in $(find . -type f -name "*.m4"); do
-                substituteInPlace $i \
-                  --replace 'test -x "$PKG_CONFIG"' 'type -P "$PKG_CONFIG" >/dev/null'
-              done
-            ''
-            + attrs.preConfigure;
-        };
+      phpAttrsOverrides = attrs: {
+        patches =
+          attrs.patches
+          or []
+          ++ prev.lib.optionals (prev.lib.versions.majorMinor args.version == "7.2") [
+            # Building the bundled intl extension fails on Mac OS.
+            # See https://bugs.php.net/bug.php?id=76826 for more information.
+            (prev.pkgs.fetchpatch {
+              url = "https://bugs.php.net/patch-display.php?bug_id=76826&patch=bug76826.poc.0.patch&revision=1538723399&download=1";
+              sha256 = "aW+MW9Kb8N/yBO7MdqZMZzgMSF7b+IMLulJKgKPWrUA=";
+            })
+          ];
+
+        configureFlags =
+          attrs.configureFlags
+          ++ prev.lib.optionals (prev.lib.versionOlder args.version "7.4") [
+            # phar extension’s build system expects hash or it will degrade.
+            "--enable-hash"
+
+            "--enable-libxml"
+            "--with-libxml-dir=${prev.libxml2.dev}"
+          ];
+
+        buildInputs =
+          attrs.buildInputs
+          ++ prev.lib.optionals (prev.lib.versionOlder args.version "7.1") [
+            prev.libxcrypt
+          ];
+
+        preConfigure =
+          prev.lib.optionalString (prev.lib.versionOlder args.version "7.4") ''
+            # Workaround “configure: error: Your system does not support systemd.”
+            # caused by PHP build system expecting PKG_CONFIG variable to contain
+            # an absolute path on PHP ≤ 7.4.
+            # Also patches acinclude.m4, which ends up being used by extensions.
+            # https://github.com/NixOS/nixpkgs/pull/90249
+            for i in $(find . -type f -name "*.m4"); do
+              substituteInPlace $i \
+                --replace 'test -x "$PKG_CONFIG"' 'type -P "$PKG_CONFIG" >/dev/null'
+            done
+          ''
+          + attrs.preConfigure;
+      };
 
       # For passing pcre2 to php-packages.nix.
-      callPackage =
-        cpFn:
-        cpArgs:
-
+      callPackage = cpFn: cpArgs:
         prev.callPackage
-          cpFn
```

Comment:
```
function
argument1
arg2
```

to me looks too much like a list of equal items so I have started indenting the arguments:

```
function
  argument1
  argument2
```

Will need to do more testing to see how it plays out in practice.